libraries/chain/include/eosio/chain/hotstuff/hotstuff.hpp

```diff
@@ -223,15 +218,14 @@ namespace eosio::chain {
       std::vector<uint8_t> _proposal_digest;
       state_t              _state { state_t::unrestricted };
       size_t               _num_finalizers {0};
-      size_t               _quorum {0};
+      std::vector<uint64_t> _finalizer_weights; // weight of each finalizer
```

Comment:
If passing in the weight, every time it needs to be passed from `block_state::aggregate_vote` to `pending_quorum_certificate::add_vote` to `pending_quorum_certificate::add_strong_vote` or `pending_quorum_certificate::add_weak_vote`. Plus we have already stored `quorum` and `num_finalizers` as members.

Reply:
> If passing in the weight, every time it needs to be passed from `block_state::aggregate_vote` to `pending_quorum_certificate::add_vote` to `pending_quorum_certificate::add_strong_vote` or `pending_quorum_certificate::add_weak_vote`. Plus we have already stored `quorum` and `num_finalizers` as members.

Still, this would make the change much simpler, and there is no reason to store a whole vector of weights, recreated every time we create a quorum certificate, just because we need to weight of the current finalizer in `add_vote`.